src/arkode/arkode_lsrkstep_impl.h

```diff
@@ -0,0 +1,140 @@
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2024, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ *---------------------------------------------------------------
+ * Implementation header file for ARKODE's LSRK time stepper
+ * module.
+ *--------------------------------------------------------------*/
+
+#ifndef _ARKODE_LSRKSTEP_IMPL_H
+#define _ARKODE_LSRKSTEP_IMPL_H
+
+#include <arkode/arkode_lsrkstep.h>
+
+#include "arkode_impl.h"
+
+#ifdef __cplusplus /* wrapper to enable C++ usage */
+extern "C" {
+#endif
+
+/*===============================================================
+  LSRK time step module data structure
+  ===============================================================*/
+
+/*---------------------------------------------------------------
+  Types : struct ARKodeLSRKStepMemRec, ARKodeLSRKStepMem
+  ---------------------------------------------------------------
+  The type ARKodeLSRKStepMem is type pointer to struct
+  ARKodeLSRKStepMemRec.  This structure contains fields to
+  perform an explicit Runge-Kutta time step.
+  ---------------------------------------------------------------*/
+typedef struct ARKodeLSRKStepMemRec
+{
+  /* LSRK problem specification */
+  ARKRhsFn fe;
+  ARKRhsFn fi;
+  ARKDomEigFn extDomEig;
+
+  /* LSRK method storage and parameters */
+  // N_Vector  temp1; /* Temp vector storage */
+  // N_Vector  temp2; /* Temp vector storage */
+  N_Vector* Fe; /* RHS vector storage */
+  N_Vector* Fi; /* RHS vector storage */
+
+  int q; /* method order               */
+  int p; /* embedding order            */
+
+  int reqstages; /* number of requested stages   */
+
+  ARKODE_LSRKMethodType LSRKmethod;
+
+  /* Counters and stats*/
+  long int nfe;            /* num fe calls       */
+  long int nfi;            /* num fi calls       */
+  long int domeignfe;      /* num fe calls for spectral dom_eig      */
+  long int ndomeigupdates; /* num of dom_eig computations   */
+  int stagemax;            /* num of max stages taken      */
+  int stagemaxlimit;       /* max allowed num of stages     */
+  int nstsig; /* num of steps that successfully used dom_eig; indicates dom_eig update when 0;  */
+
+  /* Spectral info */
+  sunrealtype lambdaR;    /* Real part of the dominated eigenvalue*/
+  sunrealtype lambdaI;    /* Imaginary part of the dominated eigenvalue*/
+  sunrealtype sprad;      /* spectral radius*/
+  sunrealtype sprmax;     /* max spectral radius*/
+  sunrealtype sprmin;     /* min spectral radius*/
+  sunrealtype domeigsfty; /* some safety factor for the user provided dom_eig*/
+  int domeigfreq; /* indicates dom_eig update after domeigfreq successful steps*/
+
+  /* Flags */
+  sunbooleantype isextDomEig; /* flag indicating user provided dom_eig */
```

Comment:
This looks equivalent to checking if `extDomEig != NULL` in which case it could be removed. Maybe having both will support future extensions

Reply:
That is true. It is just an unused variable for the time being. Should I remove it until we get to that point?